main/core/src/eval/Evaluator.scala

```diff
@@ -856,7 +856,7 @@ object Evaluator{
   case class State(rootModule: mill.define.BaseModule,
                    classLoaderSig: Seq[(Either[String, java.net.URL], Long)],
                    workerCache: mutable.Map[Segments, (Int, Any)],
-                   watched: Seq[(os.Path, Long)])
+                   watched: Seq[(() => Long, Long)])
```

Comment:
I'd love to see some `@param` ScalaDoc here. `() => Long` is way less self-expressive than `os.Path`.

Reply:
Yeah this `() => Long` thing is annoying. I'll go wrap it in a `Watchable` class or something so it's easier to read and update this PR with the updated version of Ammonite